### PR TITLE
【Hackathon 8th No.37】Add YOLO11 for PaddleYOLO

### PR DESCRIPTION
**Pull request by GreatV**
This pull request introduces the new YOLO11 architecture and its configurations. The changes include adding new configuration files, modifying existing initialization files to include YOLO11, and implementing the YOLO11 architecture along with its backbone and neck components.

### Introduction of YOLO11 Architecture:

* [`ppdet/modeling/architectures/yolo11.py`](diffhunk://#diff-1e551a20763a8110d7aa9f124e5ad04ac437b133cb2ae71d699e8d08e6f97df5R1-R114): Implemented the YOLO11 architecture, including the backbone, neck, and head components, as well as the forward pass and loss computation methods.

### Configuration Files:

* [`configs/yolo11/_base_/optimizer_100e.yml`](diffhunk://#diff-4290f5bcf2a3c74e46f442ccadffa8e6b59055a1902df07f3609e91274736527R1-R20): Added optimizer configuration with learning rate schedulers and optimizer settings for 100 epochs.
* [`configs/yolo11/_base_/optimizer_100e_high.yml`](diffhunk://#diff-e0c547d777f0a282e2bd130e86b089148c03108d1e5a9e8fa12459d3d459f650R1-R20): Added a high learning rate optimizer configuration for 100 epochs.
* [`configs/yolo11/_base_/yolo11_cspdarknet.yml`](diffhunk://#diff-5915cc499c990b19d692bb2dc1377b6baf084ef71958bb018a88e4944cbb8126R1-R35): Added base configuration for YOLO11 with CSPDarkNet backbone and YOLOv8 head.
* [`configs/yolo11/_base_/yolo11_reader.yml`](diffhunk://#diff-b1b6ca4cf1aa1dc6d7743a19fc076ecad50c21cc931f57e543f3356217d3f448R1-R45): Added data reader configuration for training, evaluation, and testing.
* [`configs/yolo11/_base_/yolo11_reader_high_aug.yml`](diffhunk://#diff-b01cf4fb2bcd63d2c414c67399996e307f7f420f25a708a29a5b2174737a4badR1-R45): Added data reader configuration with high augmentation for training, evaluation, and testing.
* [`configs/yolo11/yolo11_n_100e_coco.yml`](diffhunk://#diff-bad19303d15dcb41567b3bca8746233514b93cc0149ba7046c5318837bbc5bd2R1-R18): Added a complete configuration for training YOLO11 on the COCO dataset for 100 epochs.

### Modifications to Initialization Files:

* [`ppdet/modeling/architectures/__init__.py`](diffhunk://#diff-5584c74ce78706adb2479cd81d19e79574a658f1f134bebb73db24944d14aeffR27): Included YOLO11 in the architecture initialization file. [[1]](diffhunk://#diff-5584c74ce78706adb2479cd81d19e79574a658f1f134bebb73db24944d14aeffR27) [[2]](diffhunk://#diff-5584c74ce78706adb2479cd81d19e79574a658f1f134bebb73db24944d14aeffR41)
* [`ppdet/modeling/backbones/__init__.py`](diffhunk://#diff-6b7969c2721000d52037e737451b1cf021fa5dd66a540350c219aa9043e6cca1R32): Included YOLO11 CSPDarkNet in the backbone initialization file. [[1]](diffhunk://#diff-6b7969c2721000d52037e737451b1cf021fa5dd66a540350c219aa9043e6cca1R32) [[2]](diffhunk://#diff-6b7969c2721000d52037e737451b1cf021fa5dd66a540350c219aa9043e6cca1R51-R52)
* [`ppdet/modeling/necks/__init__.py`](diffhunk://#diff-f74050b89cbb2fef6ed2f42620ebecfda6e0de754af432ab5ed5174119a51800R24): Included YOLO11 PAFPN in the neck initialization file. [[1]](diffhunk://#diff-f74050b89cbb2fef6ed2f42620ebecfda6e0de754af432ab5ed5174119a51800R24) [[2]](diffhunk://#diff-f74050b89cbb2fef6ed2f42620ebecfda6e0de754af432ab5ed5174119a51800R35)

### Implementation of YOLO11 Components:

* [`ppdet/modeling/necks/yolo11_pafpn.py`](diffhunk://#diff-bad4124747516100f0f359e5040742fdb898a9a0a1b2050f5a3ae2032e8a74fbR1-R199): Implemented the YOLO11 Path Aggregation Feature Pyramid Network (PAFPN) for multi-scale feature fusion.

### Test Adjustments:

* [`ppdet/modeling/tests/test_architectures.py`](diffhunk://#diff-888697d73d9f39f51d2c0fd4bcdbc690a50bcfaddd85d2b44c7279a5504fb89fL55-R55): Updated the YOLOv6 configuration file path in the test setup.

### CODE REVIEW

**Comment by luotao1** [comment]
请先提交 RFC 设计文档

**Review by nemonameless — APPROVED** [review]

File: requirements.txt [hunk 1]
```diff
@@ -1,8 +1,8 @@
-numpy < 1.24
+numpy
 tqdm
 typeguard
-visualdl>=2.2.0
-opencv-python <= 4.6.0
+visualdl
+opencv-python
 PyYAML
```
Comment (Sunting78) [inline requirements.txt:6]: 这里去除版本限制，是否会影响其他模型呢？

Comment (Sunting78) [reply requirements.txt:6]: 训练完成的模型是否可以提供一份？便于进行精度复现。
